tombealby/grocery_store
# Shopping cart application containing the functionality to apply simple discounts to goods

This PR contains the code for a Shopping cart application containing the functionality to apply simple discounts to goods.

The code includes 3 test classes which should be run as JUnit tests. These 3 classes all contain `Test` in their name. `CostCalculatorTest.java` is the most important of the test classes. The class invokes the methods of the `CostCalculator` implementations. It is the `CostCalculator` implementations that provide the processing that calculates the total cost of a shopping basket.

Test instructions -
Run the following classes as JUnit tests
-  `CostCalculatorTest.java`
-  `DiscountCalculatorBuyOneGetOneFreeImplTest.java`
- `DiscountCalculatorThreeForPriceOfTwoImplTest.java`

Please note the following tasks that have not been done due to time pressures -
- **the cost of a basket of goods is displayed in pence.** The specification is to display the cost in pounds. It should be a trivial change to correct this.
- the code contains very limited tests.
- the `PriceList` class should perhaps be implemented as a singleton.
- the comments in the code should perhaps be expanded
